scripts/clear-all.cmake

```diff
@@ -270,6 +270,7 @@ unset(ENV{TEMP_FILE_DIR})
 unset(ENV{TEMP_ROOT})
 unset(ENV{TEST}) # Break Makefile OpenSSL build
 unset(ENV{TOOLCHAINS})
+unset(ENV{TVOS_DEPLOYMENT_TARGET})
```

Comment:
could you enlighten me where this file is used and how and why. I've never touched this file. I'd like to understand what this does before approving the PR :sweat_smile: 

maybe add a comment why this env-variable was added to the file (like for the `ENV{TEST}` variable two lines above